scripts/ai2-internal/run_with_beaker.py

```diff
@@ -66,56 +61,73 @@ def main(param_file: str, extra_beaker_commands: List[str]):
+        },
+        "env": env
+    }
+    config_task = {"spec": config_spec}
+    if args.name:
+        config_task["name"] = args.name
+
+    config = {
+        "tasks": [config_task]
+    }
+
+    output_path = args.spec_output_path if args.spec_output_path else tempfile.mkstemp(".yaml",
+            "beaker-config-")[1]
+    with open(output_path, "w") as output:
+        output.write(json.dumps(config, indent=4))
+    print(f"Beaker spec written to {output_path}.")
+
+    experiment_command = ["beaker", "experiment", "create", "--file", output_path]
+    if args.name:
+        experiment_command.append("--name")
+        experiment_command.append(args.name)
+
+    if args.dry_run:
+        print(f"This is a dry run (--dry-run).  Launch your job with the following command:")
+        print(f"    " + " ".join(experiment_command))
+    else:
+        print(f"Running the experiment:")
+        print(f"    " + " ".join(experiment_command))
+        subprocess.run(experiment_command)
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
 
     parser.add_argument('param_file', type=str, help='The model configuration file.')
     parser.add_argument('--name', type=str, help='A name for the experiment.')
+    parser.add_argument('--spec_output_path', type=str, help='The destination to write the experiment spec.')
+    parser.add_argument('--dry-run', action='store_true', help='If specified, an experiment will not be created.')
+    parser.add_argument('--blueprint', type=str, help='The Blueprint to use (if unspecified one will be built)')
     parser.add_argument('--desc', type=str, help='A description for the experiment.')
-    parser.add_argument('--debug', action='store_true', help='Print verbose stack traces on error.')
-    parser.add_argument('--env', action='append', help='Set environment variables (e.g. NAME=value or NAME)')
-    parser.add_argument('--mount', action='append', help='Bind a host directory (e.g. /host/path:/target/path)')
-    parser.add_argument('--source', action='append', help='Bind a remote data source (e.g. source-id:/target/path)')
+    parser.add_argument('--env', action='append', default=[], help='Set environment variables (e.g. NAME=value or NAME)')
+    parser.add_argument('--source', action='append', default=[], help='Bind a remote data source (e.g. source-id:/target/path)')
     parser.add_argument('--cpu', help='CPUs to reserve for this experiment (e.g., 0.5)')
     parser.add_argument('--gpu_count', help='GPUs to use for this experiment (e.g., 1 (default))')
     parser.add_argument('--memory', help='Memory to reserve for this experiment (e.g., 1GB)')
 
     args = parser.parse_args()
 
-    extra_beaker_commands = []
-    if args.desc:
-        extra_beaker_commands.append(f'--desc={args.desc}')
-    if args.name:
-        # Remove spaces from the name, because Beaker doesn't allow them.
-        extra_beaker_commands.append(f'--name={args.name.replace(" ", "-")}')
```

Comment:
Please don't change this behaviour, it is convenient

Reply:
Apologies--the change was a mistake (I deleted the name argument and then re-added it).